crates/trigger2/src/cli.rs

```diff
@@ -194,11 +194,18 @@ impl<T: Trigger> FactorsTriggerCommand<T> {
 
         let runtime_config = match &self.runtime_config_file {
             Some(runtime_config_path) => {
-                ResolvedRuntimeConfig::from_file(runtime_config_path, self.state_dir.as_deref())?
+                ResolvedRuntimeConfig::<TriggerFactorsRuntimeConfig>::from_file(
+                    runtime_config_path,
+                    self.state_dir.as_deref(),
+                )?
             }
             None => ResolvedRuntimeConfig::default(),
         };
 
+        runtime_config
+            .set_initial_key_values(&self.key_values)
+            .await?;
```

Comment:
May be out of scope for this PR, but I think we should probably support something like:

```toml
[key_value_store.default]
type = "none"
```

...which would then suggest doing something like:

```suggestion
        if !self.key_values.is_empty() {
            runtime_config
                .set_initial_key_values(&self.key_values)
                .await?;
        }
```

Reply:
Is this point of such a feature to ensure that a default store never gets created? If the `key-value` interface gets used with the default store is it expected to just error?

Reply:
Yep. Could be helpful if you know your target environment won't have a KV store.